### DISCUSSION TITLE
Analysis-function

### DISCUSSION THREAD

**Discussion by henfri** [XKNX/knx-integration#13] [Ideas]
Preface: I know this is a very large feature request. However, the added value is significant, so I wanted to share it.

**Step 1:**
Home Assistant (HA) already has the (possibility) to import project files and display them.
In addition, HA monitors the bus traffic. It would be incredibly useful if the view of the Group Addresses (GAs) included four additional columns:
- Current state
- Set by
- Previous state
- Set by
This would greatly assist with analysis. 
Part of this already exists in https://github.com/XKNX/knx-frontend/pull/181

Even better, though more complex since this data and view do not yet exist:
**Step 2:**
The project file also contains all devices with their communication objects (COs) and links
It might therefore be possible to display each device with its COs in HA and, as in Step 1, show the state.

[Dream mode]
**Step 3:**
One could even introduce a slider/time picker to step forward and backward incrementally.

All of this would greatly enhance the possibilities to debug the KNX-Installation with the help of HA

Greetings,
Hendrik

Reference: https://knx-user-forum.de/forum/projektforen/home-assistant/1996781-analysem%C3%B6glichkeiten-in-ha

**Comment by farmio** [comment]
I think we have all that now 🙂
